### ISSUE TITLE
ReLU on RNNs

### ISSUE THREAD

**Issue by sbhaaf** [pytorch/pytorch#1932]
Hello, I would really like the functionality to use a different activation function on the output of the RNNs. I have found ReLU more useful in classification models because, for instance, tanh output from an LSTM makes it easy for a subsequent softmax-linear layer to produce values near .999.

**Comment by ngimel** [comment]
I take it, you want it for LSTMs, because RNNs already support ReLU activations?

**Comment by sbhaaf** [comment]
Yes, and GRUs. The ReLU for the RNN implements the RNNReLUCell, but wouldn't it be more flexible if there were only an RNNCell and the activation was an argument?

**Comment by ngimel** [comment]
The API questions/suggestions go back to still unresolved #711. Adding new options to recurrent cells is not particularly hard (cudnn won't support it, but Autograd would), but until everyone is settled on a better API, it would just clutter current cells with countless options.

**Comment by sbhaaf** [comment]
I see. I look forward to seeing what you guys come up with.

**Comment by sharanry** [comment]
Hey, I am new to PyTorch. Shall I work on this?

**Comment by apaszke** [comment]
@sharanry I'm afraid that RNNs in general require a large refactor. It's probably not a right issue for a newcomer 😕

**Comment by sharanry** [comment]
@apaszke Oh :(. Could you suggest any issue I can start off with?

**Comment by apaszke** [comment]
How about #1750, #2119 or #2236?

**Comment by sharanry** [comment]
@apaszke Thank You!
